### DISCUSSION TITLE
Output dimension limit for BigTIFF image?

### DISCUSSION THREAD

**Discussion by chauraph** [libvips/libvips#3573] [Q&A] [answered]
## Bug report

**Describe the bug**
Error when outputting BigTIFF image with a huge **single dimension**. We try to merge two 2,430,000 x 400,000 images horizontally to create a 4,860,000 x 400,000 BifTIFF file. We've got the `TIFFAppendToStrip: Write error at scanline 128` error. When we try merging the images vertically (i.e. 2,430,000 x 800,000), it seems working fine.

**To Reproduce**
```
PS C:\> vips.exe join D:\bigsource.tif D:\bigsource.tif E:\data\even_bigger_output.tif[bigtiff] horizontal --vips-progress
vips.exe temp-2: 4860000 x 400000 pixels, 32 threads, 128 x 128 tiles, 640 lines in buffer
vips.exe temp-2: done in 108s
E:\data\even_bigger_output.tif: write error
windows error: The operation completed successfully.

TIFFAppendToStrip: Write error at scanline 128
wbuffer_write: write failed
```

**Expected behavior**
Outputting a 4,860,000 x 400,000 BigTIFF image to the destination path.

**Actual behavior**
See **Describe the bug** and **To Reproduce**

**Environment**
(please complete the following information)
 - OS: Windows 11 Pro Ver 22H2 
 - Vips: 8.14.2
 - RAM: 256GB
 - CPU: AMD Ryzen Threadripper PRO 5955WX

**Comment by jcupitt** [comment]
Hi @chauraph,

Wow! Ok, that's very big.

Maybe you're going over 2gb for a single strip? Have you tried writing as a tiled image?

**Comment by jcupitt** [comment]
I tried:

```
$ vips replicate k2.jpg x.tif[bigtiff] 1500 1 --vips-progress
vips temp-1: 2175000 x 2048 pixels, 32 threads, 128 x 128 tiles, 640 lines in buffer
vips k2.jpg: 1450 x 2048 pixels, 32 threads, 1450 x 16 tiles, 640 lines in buffer
vips k2.jpg: done in 0.017s          
vips temp-1: done in 11.6s          
$ tiffinfo x.tif
=== TIFF directory 0 ===
TIFF Directory at offset 0x31c824010 (13363200016)
  Image Width: 2175000 Image Length: 2048
  Resolution: 72.009, 72.009 pixels/inch
  Bits/Sample: 8
  Sample Format: unsigned integer
  Compression Scheme: None
  Photometric Interpretation: RGB color
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 3
  Rows/Strip: 128
  Planar Configuration: single image plane
```

So that's 2175000 * 3 * 128 bytes per strip (128 rows per strip), or 840mb. I guess you're going past the 32 bit limit on bytes per strip.

**Comment by jcupitt** [comment, marked as answer]
You can use `tile-height` to set rowsperstrip in strip mode:

```
$ vips replicate k2.jpg x.tif[bigtiff,tile-height=16] 1500 1
$ tiffinfo x.tif | grep Strip
  Rows/Strip: 16
```

I think that would probably fix your problem.

**Reply by kmartinez** [reply]
its fantastic some people now have as many pixels in one row as we used to in a "high resolution" image ;-)

**Reply by chauraph** [reply]
I tried again today with [tile-height=256,tile-width=256].
The image is created successfully!

TIFF Directory at offset 0x9615363932176 (8bec05a0010)
  Image Width: 4005000 Image Length: 400000
  Tile Width: 256 Tile Length: 256
  Resolution: 300, 300 pixels/inch
  Bits/Sample: 16
  Sample Format: unsigned integer
  Compression Scheme: None
  Photometric Interpretation: RGB color
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 3
  Planar Configuration: single image plane
  ICC Profile: <present>, 560 bytes
